torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -140,9 +146,15 @@ std::shared_ptr<FutureMessage> RequestCallbackImpl::processRpc(
 
       auto ownerRRef = ctx.getOrCreateOwnerRRef(rrefId, PyObjectType::get());
 
-      IValue py_ivalue = jit::toIValue(
-          PythonRpcHandler::getInstance().runPythonUDF(prc.serializedPyObj()),
-          PyObjectType::get());
+      auto& pythonRpcHandler = PythonRpcHandler::getInstance();
+      IValue py_ivalue;
+      {
+        pybind11::gil_scoped_acquire ag;
+        auto pythonUdf = pythonRpcHandler.deserialize(prc.serializedPyObj());
+        py_ivalue = jit::toIValue(
+            PythonRpcHandler::getInstance().runPythonUdf(std::move(pythonUdf)),
```

Comment:
same as above, does it need GIL here?
pythonRpcHandler.runPythonUdf()

Reply:
When I drop the GIL, I hit errors/hangs in tests. Looks like even std::move an `py::object` requires GIL?